### PR TITLE
Upgrade to terraform framework v13

### PR DESCRIPTION
**Pull request by dglsparsons**
- [V12 release notes](https://github.com/hashicorp/terraform-plugin-framework/releases/tag/v0.12.0)
- [V13 release notes](https://github.com/hashicorp/terraform-plugin-framework/releases/tag/v0.13.0)

### CODE REVIEW

File: vercel/resource_project.go [hunk 1]
```diff
@@ -105,10 +132,9 @@ At this time you cannot use a Vercel Project resource with in-line ` + "`environ
 						Sensitive:   true,
 					},
 					"id": {
-						Description:   "The ID of the Environment Variable.",
-						Type:          types.StringType,
-						PlanModifiers: tfsdk.AttributePlanModifiers{resource.UseStateForUnknown()},
```
Comment (adriancooney) [inline vercel/resource_project.go:110]: We don't need these modifiers anymore? :o

**Review by adriancooney — APPROVED** [review]
A lot cleaner! Great work.

File: vercel/data_source_alias.go [hunk 2]
```diff
@@ -6,16 +6,44 @@ import (
...
+type aliasDataSource struct {
+	client *client.Client
+}
+
+func (d *aliasDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_alias"
+}
+
+func (d *aliasDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
```
Comment (kakadiadarpan) [inline vercel/data_source_alias.go:28]: 😄

**Review by kakadiadarpan — APPROVED** [review]

File: vercel/resource_project.go (hunk 1 above)
Comment (dglsparsons) [reply vercel/resource_project.go:110]: It's actually incorrect here - the project resource doesn't bother doing an update on the environment variables, and instead just deletes any that do not match the state after, and adds new ones. 

So we can't rely on stable IDs between updates for env variables.

**Pull request merged**